### PR TITLE
Text Presets: Fix default values

### LINKED ISSUE #2041: Subheading is added with font weight as "multiple"
## Bug Description

Adding text preset "Subheading" is added with font-weight showing "multiple".
This might come from subheading being assigned a non-existing font-weight.

See the related larger issue: #1899 

## Expected Behaviour
"Subheading"  preset should use an existing font-weight.

## Steps to Reproduce

1. Add text preset: Subheading
2. See the font-weight.

## Screenshots
subheading

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### LINKED ISSUE #2665: Increase default font size
## Feature Description

This is a follow-up to #1204

Valuable discussion also in https://github.com/google/web-stories-wp/pull/1546/files#r426681693

According to Figma:

However, we cannot use **16px** as-is, because the font size needs to be responsive. We currently use **1em**, which equals about 14px. **1.2em** would be closer to 16px though, so might be worth changing.

Also, the line height probably needs to be **1.5** instead of **1.3**?

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

Improves default values for text presets, particularly font size, line height and font weight.

## Relevant Technical Choices

Cleans up text presets a little bit to add stricter prop type checking and help separate preset data and element data.

## User-facing changes

* Changes minimum font size for text presets to 1.2em / roughly 16px.  
  Screenshot 2020-06-26 at 14 43 21
* Fixes a bug where the subheading preset did have an invalid font weight of 600, whereas [Roboto](https://fonts.google.com/specimen/Roboto) only supports 500 or 700, nothing in-between
  Screenshot 2020-06-26 at 14 45 46

## Testing Instructions

1. Insert new text elements from the text presets, including the "quick action" via the ➕ icon in the tab.
1. Verify that "body text" and the element added via "quick action" show a font size of 16
1. Verify that all presets have a line  height of 1.5 when added.
1. Verify  that the "subheading" element shows "Medium" for the font weight instead of "Multiple"

---

Fixes #2665
Fixes #2041

### CODE REVIEW

**Comment by miina** [comment]
Also fixes #2041, right?

File: assets/src/edit-story/components/library/panes/text/textPresets.js [hunk 1]
```diff
@@ -31,55 +31,59 @@ const DEFAULT_ELEMENT_WIDTH = PAGE_WIDTH / 2;
...
-    width: DEFAULT_ELEMENT_WIDTH,
+    element: {
+      content: __(
+        'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
+        'web-stories'
+      ),
+      fontSize: dataFontEm(1.2),
+      lineHeight: 1.5,
+      x: DEFAULT_LEFT_MARGIN,
+      y: (PAGE_HEIGHT - dataFontEm(1.1)) / 2,
```
Comment (miina) [inline assets/src/edit-story/components/library/panes/text/textPresets.js]: ^ This should also be adjusted to the new line-height now.

File: assets/src/edit-story/components/library/panes/text/textPresets.js [hunk 2]
```diff
@@ -31,55 +31,59 @@ const DEFAULT_ELEMENT_WIDTH = PAGE_WIDTH / 2;
...
 const DEFAULT_PRESET = {
-  id: 'default-text',
   content: __('Fill in some text', 'web-stories'),
-  fontSize: dataFontEm(1),
-  font: TEXT_ELEMENT_DEFAULT_FONT,
-  width: 160,
+  fontSize: dataFontEm(1.2),
+  lineHeight: 1.5,
   x: DEFAULT_LEFT_MARGIN,
   y: (PAGE_HEIGHT - dataFontEm(1)) / 2,
```
Comment (miina) [inline assets/src/edit-story/components/library/panes/text/textPresets.js]: ^ Here, too -- should also be adjusted to the new line-height/font-size now.

**Comment by swissspidy** [comment]
> Also fixes #2041, right?

Indeed it does! Didn't realize there was a ticket for that.

File: assets/src/edit-story/components/library/panes/text/textPresets.js (hunk 2 above)
Comment (swissspidy) [reply assets/src/edit-story/components/library/panes/text/textPresets.js]: Added line height now 👍

File: assets/src/edit-story/components/library/panes/text/textPresets.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/components/library/panes/text/textPresets.js]: The new line height is already added here?

Now every preset has 1.5 line height

File: assets/src/edit-story/components/library/panes/text/textPresets.js (hunk 2 above)
Comment (swissspidy) [reply assets/src/edit-story/components/library/panes/text/textPresets.js]: Every preset has new line height

Comment (miina) [reply assets/src/edit-story/components/library/panes/text/textPresets.js]: Oh, I should have been more explicit, sorry.

What I meant was that the `y` value should be adjusted as well based on the new values. The positioning goal is to vertically center the new elements which means it's taking the page height and the element height and dividing it to two for finding the center point. However, now the element height has changed but not in the `y` value.

Comment (swissspidy) [reply assets/src/edit-story/components/library/panes/text/textPresets.js]: Ah! I see. Let me fix this then.

Comment (swissspidy) [reply assets/src/edit-story/components/library/panes/text/textPresets.js]: Done now! PTAL!

**Review by miina — APPROVED** [review]
LGTM!

**Pull request merged**